embedded_rtic/examples/sonars.rs

```diff
@@ -0,0 +1,440 @@
+        let trig3 = p0.p0_16.into_push_pull_output(gpio::Level::Low).degrade();
+        let echo3 = p0.p0_15.into_pulldown_input().degrade();
+
+        // Enable interrupts
+        //
+        // The nrf52840 is a bit strange when it comes to pin interrupts,
+        // but for now we simply connect the echo pin to the channel 1.
+
+        // Configure GPIOTE and PPI for all sonars
+        let gpiote = Gpiote::new(cx.device.GPIOTE);
+        let ppi_channels = ppi::Parts::new(cx.device.PPI);
+
+        // Sonar 1 -> Channel0
+        // Sonar 2 -> Channel1
+        // Sonar 3 -> Channel2
+        let gpiote = {
+            gpiote
+                .channel0()
+                .input_pin(&echo)
+                .toggle()
+                .enable_interrupt();
+
+            gpiote
+                .channel1()
+                .input_pin(&echo2)
+                .toggle()
+                .enable_interrupt();
+
+            gpiote
+                .channel2()
+                .input_pin(&echo3)
+                .toggle()
+                .enable_interrupt();
+
+            let mut ppi0 = ppi_channels.ppi0;
+            ppi0.set_event_endpoint(gpiote.channel0().event());
+            ppi0.set_task_endpoint(gpiote.channel0().task_out());
+            ppi0.enable();
+
+            let mut ppi1 = ppi_channels.ppi1;
+            ppi1.set_event_endpoint(gpiote.channel1().event());
+            ppi1.set_task_endpoint(gpiote.channel1().task_out());
+            ppi1.enable();
+
+            let mut ppi2 = ppi_channels.ppi2;
+            ppi2.set_event_endpoint(gpiote.channel2().event());
+            ppi2.set_task_endpoint(gpiote.channel2().task_out());
+            ppi2.enable();
+
+            gpiote.port().enable_interrupt();
+            gpiote
+        };
+
+        let (sender, receiver) = make_channel!(u32, CAPACITY);
+        let previous_time = Instant::from_ticks(0);
+
+        let (sender2, receiver2) = make_channel!(u32, CAPACITY);
+        let previous_time2 = Instant::from_ticks(0);
+
+        let (sender3, receiver3) = make_channel!(u32, CAPACITY);
+        let previous_time3 = Instant::from_ticks(0);
+
+        let times = [Instant::from_ticks(0);3];
+
+        let token = rtic_monotonics::create_nrf_timer0_monotonic_token!();
+        Mono::start(cx.device.TIMER0, token);
+
+        let _ = trigger_trampoline1::spawn();
+        let _ = trigger_trampoline2::spawn();
+        let _ = trigger_trampoline3::spawn();
+
+        (Shared { gpiote }, Local {
+            trig,
+            echo,
+            trig2,
+            echo2,
+            trig3,
+            echo3,
+            sender,
+            receiver,
+            sender2,
+            receiver2,
+            sender3,
+            receiver3,
+            previous_time,
+            previous_time2,
+            previous_time3,
+            times,
+        })
+    }
+
+    const NUM_SONARS: usize = 3; // Number of sonars
+    #[derive(Copy, Clone)]
+    enum SonarChannel {
```

Comment:
move to `src/car.rs` or something similar as this will remain the same for future versions.